sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs

```diff
@@ -363,19 +363,24 @@ public virtual async Task<Response<IReadOnlyList<MetricSeriesData>>> GetMetricSe
             IEnumerable<IDictionary<string, string>> series = options.SeriesToFilter.Select(key => key.Dimension);
             MetricDataQueryOptions queryOptions = new MetricDataQueryOptions(ClientCommon.NormalizeDateTimeOffset(options.StartTime), ClientCommon.NormalizeDateTimeOffset(options.EndTime), series);
 
-            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(MetricsAdvisorClient)}.{nameof(GetMetricSeriesData)}");
-            scope.Start();
-
-            try
-            {
-                Response<MetricDataList> response = await _serviceRestClient.GetMetricDataAsync(metricGuid, queryOptions, cancellationToken).ConfigureAwait(false);
-                return Response.FromValue(response.Value.Value, response.GetRawResponse());
-            }
-            catch (Exception e)
+            async Task<Page<MetricSeriesData>> FirstPageFunc(int? pageSizeHint)
             {
-                scope.Failed(e);
-                throw;
+                using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(MetricsAdvisorClient)}.{nameof(GetMetricSeriesData)}");
+                scope.Start();
+
+                try
+                {
+                    Response<MetricDataList> response = await _serviceRestClient.GetMetricDataAsync(metricGuid, queryOptions, cancellationToken).ConfigureAwait(false);
```

Comment:
I see `GetMetricSeriesDataOptions ` has skip and count so I imagine that will affect the pageable?
But looking at the request, it is a post one and doesn't support those properties. 
How will this work if the user set them?

Reply:
Great find. This was a mistake. I'll remove the `Skip` and `Top` properties from the options.